pyomo/core/base/units_container.py

```diff
@@ -1107,6 +1108,7 @@ def exitNode(self, node, data):
 
             return (pyomo_unit, pint_unit)
 
+        # !!THIS SEEMS LIKE A BUG!! - mrmundt #
```

Comment:
@jsiirola - There is only one spot for `format` to fill data - and `node_type` is not defined anywhere before this.

Reply:
Yes.  This should probably be
```python
raise TypeError('An unhandled expression node type: {} was encountered while retrieving the'
                ' units of expression'.format(str(type(node)), str(node)))
```

Reply:
Where would the second variable go? The one that holds `str(node)`?

Reply:
Doh.  Right:
```python
raise TypeError('An unhandled expression node type: {} was encountered while retrieving the'
                ' units of expression {}'.format(str(type(node)), str(node)))
```